liblangutil/SemVerHandler.cpp

```diff
@@ -231,6 +231,10 @@ SemVerMatchExpression::MatchComponent SemVerMatchExpressionParser::parseMatchCom
 		component.levelsPresent++;
 		if (currentChar() == '.')
 			nextChar();
+		else if (currentChar() == char(-1)) // EOF
+			break;
+		else if ('0' < currentChar() || currentChar() > '9')
```

Comment:
I think you have an error here because as is the condition is equivalent to just `'0' < currentChar()`:
```suggestion
		else if ('0' > currentChar() || currentChar() > '9')
```

I think this might be nicer for readability:
```suggestion
		else if (currentChar() < '0' || '9' < currentChar())
```

Reply:
Thanks! I think there is more work to do here. Did this before Christmas and I thought it was ready, but this logic needs to be rewritten somewhat.